src/app/shared/components/template/components/combo-box/combo-box-search/combo-box-search.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, computed, input, Input, OnInit, signal } from "@angular/core";
+import { FlowTypes } from "src/app/shared/model";
+import {
+  getStringParamFromTemplateRow,
+  IAnswerListItem,
+} from "src/app/shared/utils";
+import { ModalController } from "@ionic/angular";
+
+@Component({
+  selector: "combo-box-search",
+  templateUrl: "./combo-box-search.component.html",
+  styleUrls: ["./combo-box-search.component.scss"],
+})
+export class ComboBoxSearchComponent implements OnInit {
+  public answerOptions = input.required<IAnswerListItem[]>();
+  @Input() row: FlowTypes.TemplateRow;
+  @Input() selectedValue: string;
+
+  textTitle: string | null;
+  searchTerm = signal<string>("");
+  showSearch = computed(() => this.answerOptions().length > 8);
+  selectedItem = computed(() => {
+    if (!this.selectedValue) return null;
+    return this.answerOptions().find((item) => item.name === this.selectedValue);
+  });
+
+  filteredOptions = computed(() =>
+    this.answerOptions().filter((options) =>
+      options.text.toLowerCase().includes(this.searchTerm().toLowerCase())
+    )
+  );
+
+  constructor(private modalController: ModalController) {}
+
+  ngOnInit() {
+    this.getParams();
+  }
+
+  getParams() {
+    this.textTitle = getStringParamFromTemplateRow(this.row, "text", null);
```

Comment:
nit(non-blocking): seeing as this is the only use of the `row` input, it might be better to expose a `title` property on this sub-component, and get the value from the row from within the parent component and pass that through. That would involve adding a new property to `IComboBoxParams` and the object returned from the combo-box component's `getParams` method (`modal_title` would be a better name for the param rather than `text`, but I can see `text` is already in use)